### DISCUSSION TITLE
Android Sync / Import - Export Session ID

### DISCUSSION THREAD

**Discussion by sgr78** [rustdesk/rustdesk#6387] [Feature Request]
Hello,
I'd like to request a feature and to deposit my thoughts on how this could be Implemented..
Currently on Android Devices there is No way to sync Connection IDs (login to account is not working and not implemented), 
thus for every new session a painstaking procedure should be Implemented by typing one by one the Connection IDs every time on a new device and/or for every new app install.

I think that syncing (for now) on Android can be Implement fairly easy

1. IF the android app uses the SAME .toml configuration files like the desktop version
THEN developers could point the data store of the app to save .toml connection files 
TO \sdcard\android\data\com.carriez.flutter_hbb\files\peers  INSTEAD of \data\data\com.carriez.flutter_hbb\ 
THUS will enable users to sync desktop connection files from pc to android and vice versa quite easily.

2. Android app of rustdesk has in the config area the ability to scan QR code
IF you enable the creation of connection QR codes within this app and desktop apps for every connection ID 
THEN we could import connections using the aforementioned procedure using the QR scanner within the config menu.

3. Enable both Android and Desktop apps of rustdesk to select all connection IDs for bulk export and Implement Import function.

I have already solved the syncing between my desktop by using an SMB symlink of peers folder to a central storage 
in which all my control machines connect and get their (shared) .toml connection files, but android is another beast that I wasn't able to find a solution.

Thank you for your Hard Work!

**Comment by RaXorX** [comment]
They have no intentions of working or providing this feature sadly, so you're on your own there. On android, you might be able to connect to a smb folder over a network through mixplorer or similar file explorers, that's worth a try. I'm not sure if android will work with symlinks or not in particular, but why not first make a samba connection then try adding that through a symlink on android in the respective directory. 

https://github.com/rustdesk/rustdesk/discussions/6161

Aside from the venting a little, the part about possible alternatives should be the best that one could hope for if they don't want to create user accounts on their ends. Optionally, the exported file could also be encrypted by a password for added security. But again, doesn't seem like there are intentions to provide that.

**Reply by sgr78** [reply]
I have verified,
The android app uses the SAME .toml configuration files like the desktop version
the developers should point the data store of the app to save .toml connection files
TO \sdcard\android\data\com.carriez.flutter_hbb\app_flutter\peers 
INSTEAD of \data\data\com.carriez.flutter_hbb\app_flutter\peers
THUS will enable users to sync desktop connection files from pc to android and vice versa quite easily.

Access to \data\data\com.carriez.flutter_hbb can be granted to user 
A. On unrooted device only within app, if there is an import feature to Import .toml files 
as the app has local access rights to it's own storage space
B. On rooted devices with any File Manager MiX, Root Explorer etc

So PLEASE either -redirect- peers storage to local storage or enable import of .toml files

**Reply by RaXorX** [reply]
I would say instead of just android sync we should be requesting for a full featured sync process with all devices and not just android. I think they already provide settings sync for the paid version. Doesn't seem like that'll be granted anytime soon for free users. 

From one of the threads from what I recall the passwords were being saved pretty openly. That might be a cause for concern in syncing settings file over the connections even if it's peer to peer. Either ways. Restricting baseline features like this already speaks of the intentions of the developer(s). What I was most blown away was the license key which grants you, just the license key to use the pro code. Which works well for enterprises and companies, no per user charges unlike other remoting tools. But this is also limited in the sense that currently it cannot be used for remote administration as a service provider, this is way too simple for something like that. So use case is already limited and then limitting the features like that for free users pretty much makes it nothing too different.
